src/core/DY.Core.Trace.Manipulation.fst

```diff
@@ -115,6 +115,15 @@ let add_event e tr =
   norm_spec [zeta; delta_only [`%prefix]] (prefix #label);
   ((), Snoc tr e)
 
+val add_event_event_exists:
+  e:trace_event -> tr:trace ->
+  Lemma
+  (ensures (
+    let ((), tr_out) = add_event e tr in
+    event_exists tr_out e
+  ))
+let add_event_event_exists e tr = ()
```

Comment:
It seems this is not used anywhere, and it is not a lemma we expect the user to use, so maybe we don't need it and can remove it?

Reply:
On the other hand, it doesn't hurt to have it.
Not for the user, but for us at some later point maybe. When we add other trace entries.
But I don't really mind. So either way is fine for me.

Reply:
Ok, why not keeping it then (let's say it's a useful sanity check anyway)